lib/services/user.ts

```diff
@@ -18,18 +19,52 @@ export async function getUserWithPosts(id: string) {
   });
 }
 
-export type ModifiableUser = Pick<User, "name" | "imageUrl">;
+export type ModifiableUser = { name?: string; image?: Buffer };
 
-export async function updateUser(id: string, updatedUser: ModifiableUser) {
-  return prisma.user.update({
-    where: { id },
-    data: {
-      name: updatedUser.name,
-      imageUrl: updatedUser.imageUrl,
-    },
-  });
+export async function updateUser(
+  id: string,
+  updatedProperties: ModifiableUser
+) {
+  let imageUrl: string | undefined;
+
+  if (updatedProperties.image) {
+    imageUrl = await uploadImage(updatedProperties.image);
+  }
+
+  try {
+    const currentUser = await prisma.user.findUnique({ where: { id } });
+
+    if (!currentUser) {
+      throw new Error(`User ${id} does not exist`);
+    }
+
+    const updatedUser = prisma.user.update({
+      where: { id },
+      data: {
+        name: updatedProperties.name,
+        imageUrl: imageUrl,
+      },
+    });
+
+    if (currentUser.imageUrl) {
+      await deleteImage(currentUser.imageUrl);
```

Comment:
If there is no new image, you don't want to delete this old image

Reply:
Ah good catch, thanks!